internal/core/metadata/operators/addressable/delete.go

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package addressable
+
+import (
+	"github.com/edgexfoundry/edgex-go/internal/core/metadata/errors"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+type DeleteExecutor interface {
+	Execute() error
+}
+
+type addressDelete struct {
+	database   AddressDeleter
+	identifier string
+}
+
+// This method adds the provided Addressable to the database.
+func (op addressDelete) Execute() error {
+	// Check if the addressable exists
+	idAddr, idErr := op.database.GetAddressableById(op.identifier)
+	nameAddr, nameErr := op.database.GetAddressableByName(op.identifier)
```

Comment:
These two calls shouldn't be performed all of the time. If we're deleting an addressable by ID, the first one will succeed and the second one is redundant. Vice versa if we're deleting an addressable by Name.